### ISSUE TITLE
Ember.A(…).slice no longer exists

### ISSUE THREAD

**Issue by jamesarosen** [emberjs/ember.js#17117]
https://github.com/mharris717/ember-cli-pagination/blob/bd8f3c8e6f2465bd346e60d758e7a0ccc61ddf72/addon/divide-into-pages.js#L6-L7:
```js
const all = Ember.A(this.get('all')); 
return Ember.A(all.slice(range.start,range.end+1)); 
```

This passes on Ember beta and default, but fails on canary.

See also mharris717/ember-cli-pagination#260

**Comment by jamesarosen** [comment]
Hmm. I'm getting this in CI, but not locally.

**Comment by jamesarosen** [comment]
`Ember.A(7).slice == null`. I wonder if some race condition is causing `this.get('all')` to be something invalid.

**Comment by jamesarosen** [comment]
I logged `all` and got
```
{
  isDescriptor: true,
  enumerable: true,
  configurable: true,
  altKey: 'parent.content',
  _dependentKeys: ['parent.content'],
  __EMBER_ARRAY__ember153937986784079711557912__: true,
  '[]': {
    '$ref': '$["data"][2]'
  },
  hasArrayObservers: false,
  '@each': {
    _content: { '$ref': '$["data"][2]' }
  }
}
```
That looks like `this.get('all')` returned the `all` property rather than the _value_ of it.

My guess is that this is user-error, though I haven't quite tracked it down.

**Comment by jamesarosen** [comment]
The computed property is definitely the result of [makeLocal](https://github.com/mharris717/ember-cli-pagination/blob/9f43e32c14f024fac6d7aab45f51f651cc61e5b4/addon/computed/paged-array.js), which effectively calls
```js
PagedArray.create({
  parent: this,
  content: Ember.computed.alias('parent.conten')
})
```

I wonder whether `.create` with an `Ember.computed.alias` is the problem. Perhaps it now needs to be `.extend(…).create()`.

**Comment by jamesarosen** [comment]
It looks like this was solved in ember-cli-pagination@3.1.3.
